definitions/npm/@reduxjs/toolkit_v1.x.x/flow_v0.104.x-/test_toolkit_v1.x.x.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+import { describe, it } from 'flow-typed-test';
+import { createAction, createReducer } from '@reduxjs/toolkit';
+
+describe('@redux/toolkit', () => {
+  describe('createAction', () => {
+    it('supports being called with an action type string', () => {
+      createAction('increment');
+    });
+
+    it('supports being called with an action type string and a payload type', () => {
+      type Payload = $ReadOnly<{|
+        someData: 'The FitnessGram Pacer Test is a multistage aerobic capacity test that progressively gets...',
+      |}>;
+
+      createAction<Payload>('injectData');
+    });
+  });
+
+  describe('createReducer', () => {
+    it('supports being called with an object map', () => {
+      const state = {
+        name: 'Pelle',
+      };
+
+      const SET_NAME = 'SET_NAME';
+
+      createReducer(state, {
+        [SET_NAME]: (state, action) => {
+          state.name = action.payload.name;
+        }
+      });
+    });
+
+    it('supports being called with an builder callback', () => {
+      const state = {
+        name: 'Pelle',
+      };
+
+      const setName = createAction<string>('SET_NAME');
+
+      createReducer(state, builder => {
+        builder.addCase(setName, (state, action) => {
+          state.name = action.payload.name;
+        });
+      });
+    });
+  });
+});
```

Comment:
**suggestion:** could you add a few error assertions as well?

Reply:
The test suite does not properly run those it seems so I skipped them. I just get `Unused suppression comment` on all `// $FlowExpectedError`s, Even though the type is correct and should give error. It spams that form other libdefs too.

Reply:
Can you give an example of it not working or the issues you're getting?